pkg/util/externalmount/mount.go

```diff
@@ -0,0 +1,372 @@
+/*
+Copyright 2014 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// TODO(thockin): This whole pkg is pretty linux-centric.  As soon as we have
+// an alternate platform, we will need to abstract further.
+package externalmount
```

Comment:
Can we keep this pkg name as `mount`, and rename the internal one to something like `volume`?  The methods that are left in the internal interface are not really mount related anymore.  I know it will be confusing with `pkg/volume/util`. Maybe we can move everything into the same directory/pkg?  I'm ok with doing it in a separate change to try to keep this smaller.

Reply:
yeah, so I did create a new `pkg/util/volume`, which as you said is a bit confusing. But I think merging that with `pkg/volume/util` should be a (much desired) second step.